packages/core/contracts/financial-templates/common/FundingRateApplier.sol

```diff
@@ -94,15 +173,56 @@ abstract contract FundingRateApplier is Testable, Lockable {
         view
         returns (FixedPoint.Unsigned memory tokenDebt)
     {
-        return rawTokenDebt.mul(cumulativeFundingRateMultiplier);
+        return rawTokenDebt.mul(fundingRate.cumulativeMultiplier);
     }
 
-    function _getFundingRateStore() internal view returns (FundingRateStoreInterface) {
-        return FundingRateStoreInterface(finder.getImplementationAddress("FundingRateStore"));
+    function _getOptimisticOracle() internal view returns (OptimisticOracle) {
+        return OptimisticOracle(finder.getImplementationAddress(OracleInterfaces.OptimisticOracle));
     }
 
-    function _getLatestFundingRate() internal view returns (FixedPoint.Signed memory) {
-        return _getFundingRateStore().getFundingRateForContract(address(this));
+    function _getConfig() internal view returns (ConfigStoreInterface.ConfigSettings memory) {
+        return configStore.getCurrentConfig();
+    }
+
+    function _getLatestFundingRate() internal returns (FixedPoint.Signed memory) {
+        uint256 timestamp = fundingRate.proposalTime;
```

Comment:
Nit: this variable could be better named prehaps. When you are comparing timestamp later on it's not clear that it's the funding rate proposal time by just looking at the variable name

Reply:
Good point. There are a lot of timestamps floating around. More specific names would be helpful.

Reply:
Done.